### PR TITLE
Enhance telemetry component README

### PR DESCRIPTION
**Pull request by chrkl**
**Description**

Adapt the telemetry component's README.md file according to latest enhancements for tracing and the overall observability strategy.

Changes proposed in this pull request:

- Add documentation for configurable tracing
- Instructions to deploy Kyma with configurable tracing enabled
- Link concept documentation for observability

**Related issue(s)**

#15708

### CODE REVIEW

File: components/telemetry-operator/README.md [hunk 1]
```diff
@@ -2,15 +2,33 @@
 
 ## Overview
 
-The telemetry operator contains a logging controller that generates a Fluent Bit configuration from one or more LogPipeline and LogParser custom resources. The controller ensures that all Fluent Bit Pods run the current configuration by deleting Pods after the configuration has changed. See all [CRD attributes](apis/telemetry/v1alpha1/logpipeline_types.go) and some [examples](config/samples).
+To implement [Kyma's strategy](https://github.com/kyma-project/community/blob/main/concepts/observability-strategy/strategy.md) of moving from in-cluster observability backends to telemetry components that integrate with external backends, the telemetry operator provides APIs for configurable logging, tracing, and monitoring.
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:5]:
```suggestion
To implement [Kyma's strategy](https://github.com/kyma-project/community/blob/main/concepts/observability-strategy/strategy.md) of moving from in-cluster observability backends to a Telemetry component that integrates with external backends, the telemetry operator provides APIs for configurable logging, tracing, and monitoring.
```

File: components/telemetry-operator/README.md [hunk 2]
```diff
@@ -2,15 +2,33 @@
...
+
+The telemetry operator has been bootstrapped with [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) 3.6.0. Additional APIs can also be [added by Kubebuilder](https://book.kubebuilder.io/cronjob-tutorial/new-api.html).
+
+### Configurable Logging
+
+The logging controllers generates a Fluent Bit configuration from one or more LogPipeline and LogParser custom resources. The controllers ensure that all Fluent Bit Pods run the current configuration by restarting Pods after the configuration has changed. See all [CRD attributes](apis/telemetry/v1alpha1/logpipeline_types.go) and some [examples](config/samples).
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:11]: Either singular or plural

```suggestion
The logging controller generates a Fluent Bit configuration from one or more LogPipeline and LogParser custom resources. The controllers ensure that all Fluent Bit Pods run the current configuration by restarting Pods after the configuration has changed. See all [CRD attributes](apis/telemetry/v1alpha1/logpipeline_types.go) and some [examples](config/samples).
```

```suggestion
The logging controllers generate a Fluent Bit configuration from one or more LogPipeline and LogParser custom resources. The controllers ensure that all Fluent Bit Pods run the current configuration by restarting Pods after the configuration has changed. See all [CRD attributes](apis/telemetry/v1alpha1/logpipeline_types.go) and some [examples](config/samples).
```

File: components/telemetry-operator/README.md [hunk 3]
```diff
@@ -2,15 +2,33 @@
...
 
 For now, creating Fluent Bit Pods is out of scope of the operator. An existing Fluent Bit DaemonSet is expected.
 
 The generated ConfigMap (by default, `telemetry-fluent-bit-sections` in the `kyma-system` namespace) must be mounted to the Fluent Bit Pods and consumed by an `@INCLUDE` statement in an existing [configuration file](https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode/configuration-file). Fluent Bit parsers, file references, and environment variables are available in an additional ConfigMap or Secret.
 
 See the flags that configure all ConfigMaps, Secret and DaemonSet names in [main.go](main.go).
 
-The operator has been bootstrapped with [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) 3.6.0. Additional APIs can also be [added by Kubebuilder](https://book.kubebuilder.io/cronjob-tutorial/new-api.html).
+Further design decisions and test results are documented in the [Dynamic Logging Backend Configuration](https://github.com/kyma-project/community/tree/main/concepts/observability-strategy/configurable-logging) concept documentation.
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:19]:
```suggestion
Further design decisions and test results are documented in [Dynamic Logging Backend Configuration](https://github.com/kyma-project/community/tree/main/concepts/observability-strategy/configurable-logging).
```

File: components/telemetry-operator/README.md [hunk 4]
```diff
@@ -2,15 +2,33 @@
...
+
+### Configurable Tracing
+
+>**Configurable tracing is still in development and not active with the default Kyma settings.**
+
+The trace controller creates a [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) deployment and related Kubernetes objects from a `TracePipeline` custom resource. The collector is configured to receive traces via the OTLP and OpenCensus protocols and forwards the received traces to a configurable OTLP backend.
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:25]:
```suggestion
The trace controller creates an [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) deployment and related Kubernetes objects from a `TracePipeline` custom resource. The collector is configured to receive traces using the OTLP and OpenCensus protocols, and forwards the received traces to a configurable OTLP backend.
```

File: components/telemetry-operator/README.md [hunk 5]
```diff
@@ -2,15 +2,33 @@
...
+
+See [Dynamic Trace Backend Configuration](https://github.com/kyma-project/community/tree/main/concepts/observability-strategy/configurable-tracing) for further information.
+
+### Configurable Monitoring
+
+Configurable monitoring is not implemented yet. Future plans are documented in the [Dynamic Monitoring Backend Configuration](https://github.com/kyma-project/community/tree/main/concepts/observability-strategy/configurable-monitoring) concept.
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:31]:
```suggestion
Configurable monitoring is not implemented yet. Future plans are documented in [Dynamic Monitoring Backend Configuration](https://github.com/kyma-project/community/tree/main/concepts/observability-strategy/configurable-monitoring).
```

File: components/telemetry-operator/README.md [hunk 6]
```diff
@@ -69,3 +87,12 @@ make build-image DOCKER_PUSH_DIRECTORY=<my container repo>
 make push-image DOCKER_PUSH_DIRECTORY=<my container repo> DOCKER_TAG=latest
 kubectl -n kyma-system set image deployment telemetry-operator manager=<my container repo>/telemetry-operator:latest
 ```
+
+### Enable Tracing Controller
+
+Configurable tracing will be active when following the steps above to run the telemetry operator in your development environment. When deploying the operator with the [telemetry chart](https://github.com/kyma-project/kyma/tree/main/resources/telemetry), installing the `TracePipeline` CRD manually and setting a feature flag to enable tracing is required:
```
Comment (NHingerl) [inline components/telemetry-operator/README.md:93]:
```suggestion
To activate configurable tracing, follow the previous steps to run the telemetry operator in your development environment. When deploying the operator with the [telemetry chart](https://github.com/kyma-project/kyma/tree/main/resources/telemetry), you must install the `TracePipeline` CRD manually and set a feature flag to enable tracing:
```

File: components/telemetry-operator/README.md (hunk 2 above)
Comment (chrkl) [reply components/telemetry-operator/README.md:11]: The plural for controllers was intentional. We have a LogPipeline controller and a LogParserController.

Comment (NHingerl) [reply components/telemetry-operator/README.md:11]: alright then let's use 


```suggestion
The logging controller generates a Fluent Bit configuration from one or more LogPipeline and LogParser custom resources. The controllers ensure that all Fluent Bit Pods run the current configuration by restarting Pods after the configuration has changed. See all [CRD attributes](apis/telemetry/v1alpha1/logpipeline_types.go) and some [examples](config/samples).
```

**Review by NHingerl — APPROVED** [review]

**Pull request merged**